src/services/utilities.ts

```diff
@@ -1739,7 +1739,14 @@ export function findPrecedingToken(position: number, sourceFile: SourceFileLike,
                 if (lookInPreviousChild) {
                     // actual start of the node is past the position - previous token should be at the end of previous child
                     const candidate = findRightmostChildNodeWithTokens(children, /*exclusiveStartPosition*/ i, sourceFile, n.kind);
-                    return candidate && findRightmostToken(candidate, sourceFile);
+                    if (candidate) {
+                        // Ensure we recurse into JSDoc nodes with children.
+                        if (isJSDocCommentContainingNode(candidate) && candidate.getChildren(sourceFile).length) {

```

Comment:
Checking the children length mirrors the change in #39912 and is needed to keep its test passing.

I'd use `getChildCount` but surprise, that only accepts a `SourceFile`, and in this helper we only have `SourceFileLike`. Seems like an oops.